mne/morph.py

```diff
@@ -0,0 +1,998 @@
+        for more). Defaults to self.verbose.
+
+    Returns
+    -------
+    source_morph : instance of SourceMorph
+        The loaded morph.
+    """
+    try:
+        logger.info('loading morph...')
+        morph_data = read_hdf5(fname)
+    except IOError:
+        raise IOError('cannot locate file: ' + fname)
+
+    source_morph = SourceMorph(None)
+    source_morph.subject_from = morph_data['subject_from']
+    source_morph.subject_to = morph_data['subject_to']
+    source_morph.subjects_dir = morph_data['subjects_dir']
+    source_morph.type = morph_data['type']
+    source_morph.src_info = morph_data['src_info']
+    source_morph.morph_data = morph_data['morph_data']
+    source_morph.options = morph_data['options']
+    source_morph.vol_info = morph_data['vol_info']
+    logger.info('[done]')
+    return source_morph
+
+
+###############################################################################
+# Make Nifti1Image
+
+def _update_src_vol_info(self, vol_info):
+    """Update information needed for as_volume"""
+
+    self.src_info['shape'] = vol_info['shape']
+    self.src_info['affine'] = vol_info['affine']
+    self.src_info['zooms'] = vol_info['zooms']
+    self.src_info['zooms_mri'] = vol_info['zooms_mri']
+
+
+def _interpolate_data(stc, morpher):
+    """Interpolate source estimate data to MRI"""
+
+    n_times = stc.data.shape[1]
+    shape = morpher.src_info['shape']
+    shape3d = shape
+    shape = (n_times, shape[0], shape[1], shape[2])
+    vol = np.zeros(shape)
+
+    mri_shape3d = morpher.vol_info['mri_from']['shape']
+    mri_shape = (n_times, mri_shape3d[0], mri_shape3d[1],
+                 mri_shape3d[2])
+    mri_vol = np.zeros(mri_shape)
+    interpolator = morpher.src_info['interpolator']
+
+    mask3d = morpher.src_info['inuse'].reshape(shape3d).astype(np.bool)
+    n_vertices = np.sum(mask3d)
+
+    n_vertices_seen = 0
+    for k, v in enumerate(vol):  # loop over time instants
+        stc_slice = slice(n_vertices_seen, n_vertices_seen + n_vertices)
+        v[mask3d] = stc.data[stc_slice, k]
+
+    n_vertices_seen += n_vertices
+
+    for k, v in enumerate(vol):
+        mri_vol[k] = (interpolator * v.ravel()).reshape(mri_shape3d)
+    vol = mri_vol
+
+    vol = vol.T
+
+    affine = morpher.src_info['affine']
+    affine = np.dot(morpher.src_info['mri_ras_t'], affine)
+    affine[:3] *= 1e3
+
+    try:
+        import nibabel as nib  # lazy import to avoid dependency
+    except ImportError:
+        raise ImportError("nibabel is required to save volume images.")
+
+    header = nib.nifti1.Nifti1Header()
+    header.set_xyzt_units('mm', 'msec')
+    header['pixdim'][4] = 1e3 * stc.tstep
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+        img = nib.Nifti1Image(vol, affine, header=header)
+
+    return img
+
+
+def stc_as_volume(stc, morph, shape=None, affine=None, zooms=None, hdr=None,
+                  fname=None, mri_resolution=False):
```

Comment:
does it have to be a public function? can you just support an API with the morph object?